Code/Tools/SceneAPI/SceneCore/Events/AssetImportRequest.cpp

```diff
@@ -55,74 +55,70 @@ namespace AZ
                 {
                     AZStd::string policy;
                     instance->GetPolicyName(policy);
                     AZ_TraceContext("PolicyName", policy.c_str());
 
-                    if (!m_previousSet && m_scene->GetManifest().GetEntryCount() > 0)
+                    if (ManifestInstancesAreEqual(m_previousManifest, m_scene->GetManifest()))
                     {
-                        return;
+                        AZ_TracePrintf(AZ::SceneAPI::Utilities::LogWindow, "UpdateManifest(%d): No Changes \n", m_reportNumber);
                     }
-
-                    auto saveToJsonOutcome = m_scene->GetManifest().SaveToJsonDocument();
-                    if (!saveToJsonOutcome.IsSuccess())
+                    else
                     {
-                        AZ_Error(AZ::SceneAPI::Utilities::ErrorWindow, false,
-                            "UpdateManifest(%d) error: %.*s",
-                            m_reportNumber,
-                            AZ_STRING_ARG(saveToJsonOutcome.GetError()));
-                        return;
+                        AZ_TraceContext("Old manifest rule count", m_previousManifest.GetEntryCount());
+                        AZ_TraceContext("New manifest rule count", m_scene->GetManifest().GetEntryCount());
+                        AZ_TracePrintf(AZ::SceneAPI::Utilities::LogWindow, "UpdateManifest(%d): Updated \n", m_reportNumber);
+                        m_previousManifest = m_scene->GetManifest();
                     }
+                }
 
-                    if (m_previous == saveToJsonOutcome.GetValue())
+                bool ManifestInstancesAreEqual(const Containers::SceneManifest& lhs, const Containers::SceneManifest& rhs) const
+                {
+                    if (lhs.IsEmpty() && rhs.IsEmpty())
                     {
-                        AZ_TraceContext("UpdateManifest", "No Changes");
-                        AZ_TracePrintf(AZ::SceneAPI::Utilities::LogWindow, "UpdateManifest(%d): No Changes \n", m_reportNumber);
-                        return;
+                        return true;
                     }
-                    m_previous = {};
-                    m_previousSet = false;
 
-                    rapidjson::StringBuffer buffer;
-                    rapidjson::Writer<rapidjson::StringBuffer> writer(buffer);
-                    if (saveToJsonOutcome.GetValue().Accept(writer))
+                    if (lhs.GetEntryCount() == rhs.GetEntryCount())
                     {
-                        AZ_TraceContext("Update Manifest", buffer.GetString());
-                        AZ_TracePrintf(AZ::SceneAPI::Utilities::LogWindow, "UpdateManifest(%d): Updated \n", m_reportNumber);
+                        auto rhsValueStorage = rhs.GetValueStorage();
+                        for (auto index = 0; index < lhs.GetEntryCount(); ++index)
+                        {
+                            const auto lhsObject = lhs.GetValue(index);
```

Comment:
Should this be const auto&?

Reply:
the index is really a size_t, but sure it can be